### PR TITLE
Generalize cloud run execution

### PR DESCRIPTION
**Pull request by chrismaree**
This PR refactors the express restful API used in the cloud run process execution to enable any arbitrary command to be fed in along with any arbitrary environment variables. This PR also changes the express API from a `GET` to a `POST` to facilitate the body being passed along with the request.

### CODE REVIEW

File: reporters/cloud-run-scripts/CloudResponse.js [hunk 1]
```diff
@@ -1,24 +1,48 @@
...
+    throw "Missing cloudRunCommand";
+  }
 
-  let reporterScript = exec("/bin/bash ../reporters/cloud-run-scripts/SendSlackReport.sh", (error, stdout, stderr) => {
-    console.log(stdout);
-    console.log(stderr);
+  // Iterate over the provided environment variables and ensure that they are all strings. This enables json configs
+  // to be passed in the req body and then set as environment variables in the child_process as a string
+  let processedEnvironmentVariables = {};
+  Object.keys(req.body.environmentVariables).forEach(key => {
```
Comment (mrice32) [inline reporters/cloud-run-scripts/CloudResponse.js:18]: nit: what happens if `environmentVariables` doesn't exist in the body?

File: reporters/cloud-run-scripts/CloudResponse.js [hunk 2]
```diff
@@ -1,24 +1,48 @@
...
+    processedEnvironmentVariables[key] =
+      typeof req.body.environmentVariables[key] == "string"
+        ? req.body.environmentVariables[key]
+        : JSON.stringify(req.body.environmentVariables[key]);
```
Comment (mrice32) [inline reporters/cloud-run-scripts/CloudResponse.js:22]: Took me a minute to get what this was doing -- very slick :)

File: reporters/cloud-run-scripts/CloudResponse.js [hunk 3]
```diff
@@ -1,24 +1,48 @@
...
-      console.error(`exec error: ${error}`);
-      res.status(400).send({
-        message: error
-      });
-    } else {
-      console.log("Daily report generated");
-      res.send("Daily Report sent!");
+  // Run the command from the request body. Note this assumes that the process is running from the /core directory.
+  // Include the environment variables. Having both ...process.env and ...processedEnvironmentVariables acts to combined
+  // the existing env variables with those passed in from the req.
```
Comment (mrice32) [inline reporters/cloud-run-scripts/CloudResponse.js:27]: This is a very nice feature. Gives the user flexibility to provide env variables at the external process level or in the request.

File: reporters/cloud-run-scripts/CloudResponse.js (hunk 1 above)
Comment (chrismaree) [reply reporters/cloud-run-scripts/CloudResponse.js:18]: That's a good point. it should be enforced in the forEach I have updated accordingly.

File: reporters/cloud-run-scripts/CloudResponse.js (hunk 2 above)
Comment (chrismaree) [reply reporters/cloud-run-scripts/CloudResponse.js:22]: ye, it's a bit convoluted. I added a comment to make it clearer :)

**Pull request merged**